### PR TITLE
SB-16: Sermon View UI

### PR DESCRIPTION
**Pull request by eccentricnode**
## Feature/Problem

Sermon View UI

## Solution

N/A

## Areas of Impact

N/A

## UI Images

Annotation 2019-12-05 123727

## Manual Testing

- Verify CI passes

### CODE REVIEW

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs';
 
 @Component({
   selector: 'app-sermon-view',
   templateUrl: './sermon-view.component.html',
   styleUrls: ['./sermon-view.component.scss']
 })
 export class SermonViewComponent implements OnInit {
+  // sermon$: Observable<Sermon>;
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts]: Why did you comment out the `sermon$` Observable?

We should have this, somewhat ready to use once we have data.
Hardcoding things in the template will not help with that.

Can you update to include the mocked out Observable, please?

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html [hunk 2]
```diff
@@ -1 +1,65 @@
-<p>sermon-view works!</p>
+<button mat-icon-button>
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html]: We need to mock this out as much as possible. Everything is hard coded here.

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html [hunk 3]
```diff
@@ -1 +1,65 @@
...
+      webkitallowfullscreen
+      mozallowfullscreen
+      allowfullscreen
+    ></iframe>
+  </div>
+</div>
+
+<div class="container mt-2">
+  <div class="d-flex flex-column align-items-start">
+    <mat-card-title style="color: #34587c;">
```
Comment (aklukas) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html]: No inline styles and this color should be available in the theme.

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html [hunk 4]
```diff
@@ -1 +1,65 @@
...
+
+<div class="container">
+  <div class="d-flex justify-content-between">
+    <div class="align-self-center">
+      <mat-card-subtitle class="fs-16">
+        Ron Peterson | Nov 24, 2019
+      </mat-card-subtitle>
+    </div>
+
+    <div class="align-self-end" style="color: #7b7b7b;">
```
Comment (aklukas) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html]: Same here with the inline styles and color

File: client/package.json [hunk 5]
```diff
@@ -43,6 +43,10 @@
     "@angular/platform-browser": "^8.2.0",
     "@angular/platform-browser-dynamic": "^8.2.0",
     "@angular/router": "^8.2.0",
+    "@ngrx/effects": "^8.5.2",
```
Comment (aklukas) [inline client/package.json]: None of this NGRX stuff should be part of this PR

File: client/yarn.lock [hunk 6]
```diff
@@ -1047,6 +1047,26 @@
     "@types/istanbul-reports" "^1.1.1"
     "@types/yargs" "^13.0.0"
 
+"@ngrx/effects@^8.5.2":
```
Comment (aklukas) [inline client/yarn.lock]: Same here

**Comment by aklukas** [comment]
Please review the mocks again for this. The back arrow is not the right size, the line between the speaker's name and the date isn't the right color, the line below it isn't the right color either. Also, the save, listen and notes buttons need work, the font is too big for the words and the icons. The title and all the content below the video should line up with the left side of the video too.

File: client/package.json (hunk 5 above)
Comment (eccentricnode) [reply client/package.json]: removed

File: client/yarn.lock (hunk 6 above)
Comment (eccentricnode) [reply client/yarn.lock]: removed

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts (hunk 1 above)
Comment (eccentricnode) [reply client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts]: Do we need to setup interpolation for the template, like on the sermon title, date, etc.?  That is part of the concern right?

Comment (eccentricnode) [reply client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts]: So, does the interpolation in the html file need to setup or no?

**Comment by eccentricnode** [comment]
> Look at the tags again as well, they aren't the right height either. 

For the tags, I am using Material mat-chip. I tried to dig through the find how to change the height, and I couldn't find anything. Material does provide a solution, using a mat-basic-chip element, but it removes basically all the styling. When I asked Victor, he told me to leave it as is.

**Comment by eccentricnode** [comment]
I have an updated screenshot on the PR. I cannot get spacing right between the line and the minister name and the date. As you can see, in the sermon-view template, I had 
Annotation 2019-12-05 123727

**Comment by eccentricnode** [comment]
Updated Screenshot
Annotation 2019-12-05 123727

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts [hunk 7]
```diff
@@ -1,15 +1,58 @@
 import { Component, OnInit } from '@angular/core';
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts]: Just a little nit-picky thing. Don't have to update to merge, but try separating imports like so:

```typescript
// angular imports
import { Component, OnInit } from '@angular/core';

// third-party imports
import { Observable } from 'rxjs';

// local imports
import { Model } from '@localapp-model';
```

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts [hunk 8]
```diff
@@ -1,15 +1,58 @@
...
+        {
+          id: "8f85c0bb-f07b-4a93-8fcf-ee64d3e0704e",
+          speaker: {
+            first_name: "Ron",
+            last_name: "Peterson",
+            position: "Pastor",
+            church_name: "Evening Light Fellowship"
+          }
+        }
+      ]
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.ts]:
```suggestion
      ],
      media: [
          {
            type: "VIDEO",
             url: "https://player.vimeo.com/video/375272866"
          }
        ]
```

File: client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html [hunk 9]
```diff
@@ -1 +1,58 @@
-<p>sermon-view works!</p>
+<div *ngIf="sermon$ | async as sermon">
+  <button mat-icon-button (click)="goBack()">
+    <mat-icon class="back-arrow" color="accent">arrow_back</mat-icon>
+  </button>
+
+  <div class="container-fluid">
+    <div class="d-flex justify-content-center">
+      <iframe
+        [src]="sanitizeUrl(sermon?.vimeoUrl)"
```
Comment (VicAv99) [inline client/apps/user-portal/src/app/sermons/sermon-view/sermon-view.component.html]: - update this to get the video off of the "media" array in the model I suggested.
- remove the "vimeoUrl".
- make sure this checks if the media is a video, before giving it the URL.

**Pull request merged**